src/test/java/it/gov/pagopa/pu/bff/connector/OrganizationClientImplTest.java

```diff
@@ -0,0 +1,105 @@
+package it.gov.pagopa.pu.bff.connector;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+
+import it.gov.pagopa.pu.p4pa_organization.controller.ApiClient;
+import it.gov.pagopa.pu.p4pa_organization.controller.generated.BrokerEntityControllerApi;
+import it.gov.pagopa.pu.p4pa_organization.controller.generated.OrganizationSearchControllerApi;
+import it.gov.pagopa.pu.p4pa_organization.dto.generated.EntityModelBroker;
+import it.gov.pagopa.pu.p4pa_organization.dto.generated.EntityModelOrganization;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+
+@ExtendWith(MockitoExtension.class)
+class OrganizationClientImplTest {
+
+  @Mock
+  private RestTemplateBuilder restTemplateBuilderMock;
+
+  @Mock
+  private RestTemplate restTemplateMock;
+
+  @Mock
+  private BrokerEntityControllerApi brokerEntityControllerApiMock;
+  @Mock
+  private OrganizationSearchControllerApi organizationSearchControllerApiMock;
+
+  @InjectMocks
+  private OrganizationClientImpl organizationClientMock;
```

Comment:
This is not a mock, this is the tested class

Reply:
updated